.github/workflows/test.yml

```diff
@@ -24,6 +24,17 @@ jobs:
       run: |
         pip install -U pip
         pip install -e '.[dev]'
-    - name: Test
+    - name: Test [wrapper]
       run: |
-        python -m unittest test/*.py
+        python -m unittest test/test_wrapper.py -v
+    - name: Test [all]
+      run: |
+        python -m unittest -v |& tee log.txt
+        grep '^FAIL:' log.txt > fails.txt
+        # if all failures is not raised in `test.test_wrapper`:
+        if [ "$(grep 'test.test_wrapper' fails.txt | wc -l)" -ne "$(wc -l < fails.txt)" ]; then
```

Comment:
If we don't expect some tests to pass would it not be better to just remove them or skip them?

If the endpoints are unreliable it may be better to rather investigate something like testcontainers and run containers for the tests or pytest-recording.

Reply:
Also another option is to use pytest markers (e.g. `@pytest.mark.unreliable`), then mark the tests that are unreliable, and then run:


```bash
pytest -m 'not unreliable'
## Below will ignore the exit code
pytest -m 'unreliable' || :
```

But regardless, I think we should have a decent answer as to why we are running tests if we don't expect them to pass, and see if there is not some pattern in pytest that is appropriate for dealing with them.

Reply:
OK, I understood.

Reply:
It's fine for now, we can consider it as part of a move to pytest later.